### PR TITLE
feat: improve UI on approval swap step flow

### PR DESCRIPTION
**Pull request by ChefBingbong**
### Summary
🧪🗑️🚧

This pull request introduces several changes to the swap feature of the PancakeSwap frontend, mainly to improve the user interface and the error handling. It refactors the `ConfirmSwapModal` component and the `SwapCommitButton` component to use a new variable `isPendingError` to handle different modal states and titles. It also adds animation effects and dynamic content to the `ApproveModalContent` and the `SwapTransactionReceiptModalContent` components using a new custom hook `useUnmountingAnimation`. Additionally, it fixes a bug in the `trade.test.ts` file and removes some redundant or unused code from various files.

> _We're swapping tokens in the dark_
> _With `vitest` and `hardhat` we spark_
> _We animate the modals with a hook_
> _We refactor and debug, no bug we overlook_

### Walkthrough
*  Add `isPendingError` prop and logic to `ConfirmSwapModal` and `SwapCommitButton` components to handle errors in approval or permitting process ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dR44), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL226-R221), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL256-R257), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-d80ee63cb0033c8371bfe572adb8191b01ba0b78bfe7dd5e566f37ad334ba234L110-R110), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-d80ee63cb0033c8371bfe572adb8191b01ba0b78bfe7dd5e566f37ad334ba234R257))
*  Refactor `ConfirmSwapModal` component to use `usePendingSwapTitle` function and `ApproveModalContent` component to render different modal contents based on confirm modal state and trade input amount currency symbol ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dR55-R65), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dR101-R104), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL97-R136))
*  Remove unused or duplicate props and imports from `ConfirmSwapModal` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dR1), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL14), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL23-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL59-R77), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-c83e4eec2c74b991702f46350b3eee8c029e14df4a79fae45f480542a806111dL218-L219))
*  Add animation effects to `ApproveModalContent` and `SwapTransactionReceiptModalContent` components using `useUnmountingAnimation` hook and `StepTitleAnimationContainer` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-44741df1af3f8a528061cf8d254f354b0d418c3c2014891d6f2e3a5993c0d0eaL2-R88), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-44741df1af3f8a528061cf8d254f354b0d418c3c2014891d6f2e3a5993c0d0eaL16-R165), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-3ef0d4f7160946928eb7fadc5f53ebce23f5327db8429a8ceadee47f59440253R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-3ef0d4f7160946928eb7fadc5f53ebce23f5327db8429a8ceadee47f59440253L10-R21), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-3da8000d6ef684592f558a75c72a3a834575b3e171bb8ccc9400217cd9392126R1-R79))
*  Replace `test:all` script with `test` script in `package.json` file of `universal-router-sdk` package and use `vitest` tool instead of `hardhat` tool for testing ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-f6540822464dc0df575ca5e40418c32c63bce949b2fdb8fced51c4be7d3995f7L33-R33))
*  Replace `hexToString` function with `BigInt` constructor in `trade.test.ts` file of `universal-router-sdk` package to fix test errors ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL74-R74), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL101-R101), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL129-R129), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL161-R161), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL209-R209), [link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eL270-R273))
*  Remove redundant condition from `if` statement in `PancakeSwapTrade` class of `pancakeswap.ts` file of `universal-router-sdk` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-566cd8974b99b34b5fdc1c4a109c3da8754c475cba38788149605d459e291b05L137-R137))
*  Comment out export of `PendingModalContent` component from `index.ts` file of `widgets-internal` package, which is no longer used ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-ecbc17cd033ad1b64886a1b0e55a0e0f96fb9676ab16a12b61bfe60faa944546R16))
*  Add commented out code for debugging purposes in `trade.test.ts` file of `universal-router-sdk` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/8008/files?diff=unified&w=0#diff-97f69569125e6d51bbd9cb2374077f694d1f48672578ba870b20398c8cc1501eR302-R303))

### CODE REVIEW

File: apps/web/src/views/Swap/V3Swap/containers/ConfirmSwapModal.tsx [hunk 1]
```diff
@@ -88,60 +98,42 @@ export const ConfirmSwapModal = memo<InjectedModalProps & ConfirmSwapModalProps>
     onDismiss?.()
   }, [customOnDismiss, onDismiss])
 
+  const stepContents = usePendingSwapTitle({
+    trade,
+  })
+
   const topModal = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (ChefBingbong) [reply apps/web/src/views/Swap/V3Swap/containers/ConfirmSwapModal.tsx:105]: yeah willl rebase and mrmoize where can

File: packages/widgets-internal/swap/ApproveModalContent.tsx [hunk 2]
```diff
@@ -1,50 +1,168 @@
...
-            <>
-              <TooltipText ml="4px" fontSize="14px" color="textSubtle" ref={targetRef}>
-                {t('Pancakeswap AMM')}
-              </TooltipText>
-              {tooltipVisible && tooltip}
-            </>
-          )}
-        </Flex>
-      </AutoColumn>
+      {pendingModalSteps.map((step: PendingConfirmModalState) => {
```
[the remark this answers is not in the document]
Comment (ChefBingbong) [reply packages/widgets-internal/swap/ApproveModalContent.tsx]: good catch yeah changing it now

File: apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx [hunk 3]
```diff
@@ -75,8 +75,8 @@ export const SwapCommitButton = memo(function SwapCommitButton({
     [Field.INPUT]: { currencyId: inputCurrencyId },
     [Field.OUTPUT]: { currencyId: outputCurrencyId },
   } = useSwapState()
-  const inputCurrency = useCurrency(inputCurrencyId)
-  const outputCurrency = useCurrency(outputCurrencyId)
+  const inputCurrency = useCurrency(inputCurrencyId) as Currency
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx:78]: why are we using type assertion here?

Comment (ChefBingbong) [reply apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx:78]: no specific reason i can remove. i think i added ages back but yep should be removed

**Pull request merged**